lib/smooth/application/network/http/regular/MIMEParser.cpp

```diff
@@ -112,6 +150,176 @@ namespace smooth::application::network::http::regular
+                    {
+                        get_more_data = true;
+                    }
+                }
+                else if (status == Status::Headers)
+                {
+                    auto p = std::search(data.cbegin(), data.cend(), crlf_double.cbegin(), crlf_double.cend());
+
+                    if (p != data.cend())  // end of headers found
+                    {
+                        auto [new_start_of_content, headers,
+                              content_disposition] = consume_headers(data.cbegin(), p + 4);
+                        id = content_disposition["name"];
+                        filename = content_disposition["filename"];
+
+                        // Log::info("dispo", "name: {}; filename: {}", content_disposition["name"],
+                        //         content_disposition["filename"]);
+                        data.erase(data.begin(), p + 4);  // also erase 2x crlf
+                        status = Status::Data;
+                        first_part = true;
+                    }
+                    else
+                    {
+                        get_more_data = true;
+                    }
+                }
+                else if (status == Status::Data)
+                {
+                    auto b = std::search(data.cbegin(), data.cend(), bound.cbegin(), bound.cend());
+
+                    if ((data.size() > chunksize + end_boundary.size()) || (b != data.cend()))  // got something to
+                                                                                                // write
+                    {
+                        while (std::distance(data.cbegin(), b - 2) > chunksize)
+                        {
+                            content_callback(id, filename, data.cbegin(), data.cbegin() + chunksize, first_part, false);
+                            data.erase(data.begin(), data.begin() + chunksize);
+                            first_part = false;
+                            b = std::search(data.cbegin(), data.cend(), bound.cbegin(), bound.cend());
+                        }
+
+                        if (b != data.cend())
+                        {
+                            content_callback(id, filename, data.cbegin(), b - 2, first_part, true);  // first_b-2 to
+                                                                                                     // avoid additional
+                                                                                                     // crlf in file end
+                            // std::vector<uint8_t>::iterator e;
+                            // std::advance (e, std::distance<std::vector<uint8_t>::const_iterator>(e, b ) );
+                            // Log::info("end of file",".");
+                            // log_data(data.begin(), e);
+                            // Log::info("the end",".");
+                            status = Status::Headers;
+                            data.erase(data.begin(), b - 2);
```

Comment:
I assume the magic two here is the same as the one above?

Since `b -2`is used twice, lets do something like `const auto end = b - 2` and use `end` instead. Also, lets use the constant mentioned above.

Reply:
o.k. fixed